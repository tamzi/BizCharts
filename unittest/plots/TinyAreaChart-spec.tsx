import React from 'react';
import TinyAreaChart from '../../src/plots/TinyAreaChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
  { year: '1991', value: 3 },
  { year: '1992', value: 4 },
  { year: '1993', value: 6 },
  { year: '1994', value: 6 },
  { year: '1995', value: 8 },
  { year: '1996', value: 9 },
  { year: '1997', value: 7 },
  { year: '1998', value: 9 },
  { year: '1999', value: 13 },
];

describe('Plots-TinyAreaChart', () => {
  test('堆叠面积图-旧版', () => {
    let chart = null;
    render(<TinyAreaChart
      data={MOCK_DATA}
      yField="value"
      xField="label"
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
})
