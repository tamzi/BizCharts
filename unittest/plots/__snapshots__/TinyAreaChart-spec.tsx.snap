// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-TinyAreaChart 堆叠面积图-旧版 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "areaStyle": Object {
    "fillOpacity": 0.6,
  },
  "autoFit": true,
  "color": "l(90) 0:#E5EDFE 1:#ffffff",
  "data": Array [
    3,
    4,
    6,
    6,
    8,
    9,
    7,
    9,
    13,
  ],
  "line": Object {
    "color": "#5B8FF9",
    "size": 1,
  },
  "tooltip": Object {
    "containerTpl": "<div class=\\"g2-tooltip\\"><div class=\\"g2-tooltip-list\\"></div></div>",
    "crosshairs": Object {
      "type": "x",
    },
    "customContent": [Function],
    "domStyles": Object {
      "g2-tooltip": Object {
        "fontSize": "10px",
        "padding": "2px 4px",
      },
    },
    "itemTpl": "<span>{value}</span>",
    "shared": true,
    "showCrosshairs": true,
    "showMarkers": false,
    "showTitle": false,
  },
}
`;
